### PR TITLE
Version Bump

### PR DESCRIPTION
**Pull request by MattToast**
Bump the SmartSim version `0.4.1` -> `0.4.2` in anticipation of release. Minor updates to docs/tutorials regarding `Model.colocate_db` -> `Model.colocate_db_tcp`

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -11,25 +11,34 @@ Jump to :ref:`SmartRedis Changelog <changelog>`
...
-This section details changes made in the development branch that have not yet been applied to a released version of the SmartSim library.
+This release of SmartSim had a focus on polishing and extending exiting
+features already provided by SmartSim. Most notably, this release provides
+support to allow users to colocate their models with an orchestrator using
+Unix domain sockets and support for launching models as batch jobs.
 
-Description
+Additionally SmartSim has updated its tool chains to provide a better user
+experience. Most notably, SmarSim can now provide Python 3.10 support. Furthermore,
+by bumping SmartSim now utilizes  SmartRedis's aggregation lists to streamline
```
Comment (al-rigazzi) [inline doc/changelog.rst:28]: Possibly because of my low English level, the sentence "by bumping SmartSim now utilizes..." sounds a little strange. Plus there is a double space before "SmartRedis". Could you rephrase this?

File: smartsim/experiment.py [hunk 2]
```diff
@@ -494,6 +494,13 @@ def create_model(
         useful for performant online inference or processing
         at runtime.
 
+        New in 0.4.2 ``Model`` instances can now be co-located with
+        an Orchestrator database over either TCP or UDS using the
+        ``Model.colocate_db_tcp`` or ``Model.colocate_db_uds`` method
+        respectively. The original ``Model.colocate_db`` method
```
Comment (al-rigazzi) [inline smartsim/experiment.py:500]: Should we say it will be deprecated?

**Review by al-rigazzi — COMMENTED** [review]
Minor requests for a couple of doc sentences, otherwise looks great! Thanks for doing this and kudos for catching the outdated colocated doc

File: smartsim/experiment.py (hunk 2 above)
Comment (MattToast) [reply smartsim/experiment.py:500]: Probably worth mentioning, I'll add it!

**Review by al-rigazzi — APPROVED** [review]
LGTM!

**Review by ashao — APPROVED** [review]
Looks good!
